cpp/ql/test/library-tests/dataflow/dataflow-tests/test.cpp

```diff
@@ -1046,6 +1046,14 @@ void memset_test(char* buf) { // $ ast-def=buf ir-def=*buf
 	sink(*buf); // $ ir MISSING: ast
 }
 
+void *realloc(void *, size_t);
+
+void test_realloc() {
+	int *src = indirect_source();
+	int *dest = (int*)realloc(src, sizeof(int));
+	sink(*dest); // $ ir, MISSING: ast
+}
```

Comment:
I expected the same test here as for taint tracking. Why did you add a different test?

Reply:
For the other test having a struct with a field is necessary to tease out the effect that the change in this PR has on taint tracking (or it is at least one way to do so). That complexity is not needed here, where this simpler test is enough to see that `realloc` carries data-flow through it.

The [test query](https://github.com/paldepind/codeql/blob/d5948d2d95f9b0f25416cfcf8428487533622e93/cpp/ql/test/library-tests/dataflow/dataflow-tests/test.ql#L3) uses [`AstFlow`](https://github.com/paldepind/codeql/blob/d5948d2d95f9b0f25416cfcf8428487533622e93/cpp/ql/test/library-tests/dataflow/dataflow-tests/TestBase.qll#L51) which only uses data-flow. So if `realloc` had not been modeled as a data-flow function the path would have been cut off at L1054.

Reply:
> The [test query](https://github.com/paldepind/codeql/blob/d5948d2d95f9b0f25416cfcf8428487533622e93/cpp/ql/test/library-tests/dataflow/dataflow-tests/test.ql#L3) uses [`AstFlow`](https://github.com/paldepind/codeql/blob/d5948d2d95f9b0f25416cfcf8428487533622e93/cpp/ql/test/library-tests/dataflow/dataflow-tests/TestBase.qll#L51) which only uses data-flow. So if `realloc` had not been modeled as a data-flow function the path would have been cut off at L1054.

Ast and IR data flow, to be exact 😄